src/Windows/Avalonia.Win32/Input/Imm32InputMethod.cs

```diff
@@ -324,6 +324,11 @@ public void HandleCompositionStart()
             if (IsActive)
             {
                 Client.SetPreeditText(null);
+
+                if (Client.SupportsSurroundingText && Client.SurroundingText.AnchorOffset != Client.SurroundingText.CursorOffset)
+                {
+                    KeyPress(Key.Delete);
```

Comment:
Simulating actions via sending events feels a bit like a hack. Could we issue a text input event with empty text instead? Or perhaps introduce a separate method while we still can?

Reply:
The Mac backend makes the exact same thing. The input context sends a delete key and we process it. But y we could just send an TextInputEvent with an empty string.

Reply:
Same for iOS. Ideally deletion could be processed by the TextInputMethodClient so we have a shared implementation for all platforms.